locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -1551,6 +1552,37 @@ public Template buildTemplate(ComputeService computeService, ConfigBag config) {
         return template;
     }
 
+    protected String toStringNice() {
+        Entities.dumpInfo(this);
+        String s = config().get(ORIGINAL_SPEC);
+        if (Strings.isBlank(s)) s = config().get(NAMED_SPEC_NAME);
+        if (Strings.isBlank(s)) s = config().get(FINAL_SPEC);
+        if (Strings.isBlank(s)) s = getDisplayName();
+        
+        String s2 = "";
+        String provider = getProvider();
```

Comment:
Here up to ~line 1581 would be much clearer if `Strings.isNonBlank(s)` were checked once and the whole block skipped if not.